ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/LogWithMetadata.java

```diff
@@ -76,9 +76,10 @@ public static List<LogWithMetadata> generate(
 
   public static List<LogWithMetadata> generate(
       final Block block, final List<TransactionReceipt> receipts, final boolean removed) {
-    final List<LogWithMetadata> logsWithMetadata = new ArrayList<>();
+    final int size = receipts.size();
```

Comment:
This generate method could just call your new generate method deriving the arguments

Reply:
Done